tests/com/magento/idea/magento2plugin/actions/generation/generator/DataModelInterfaceGeneratorTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.actions.generation.generator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.magento.idea.magento2plugin.actions.generation.NewDataModelAction;
+import com.magento.idea.magento2plugin.actions.generation.data.DataModelInterfaceData;
+
+public class DataModelInterfaceGeneratorTest extends BaseGeneratorTestCase {
+    /**
+     * Tests for generation of a Magento 2 Data Model Interface.
+     */
+    public void testGenerateDataModelInterface() {
+        final Project project = myFixture.getProject();
+        final DataModelInterfaceData interfaceData = new DataModelInterfaceData(
+                "Foo\\Bar\\Api\\Data",
+                "SampleInterface",
+                "Foo_Bar",
+                "Foo\\Bar\\Api\\Data\\SampleInterface",
+                "SAMPLE_PROPERTY;sample_property;string;SampleProperty;sampleProperty"
```

Comment:
Hi.
I would use an array of Property objects. It would make a code cleaner. What do you think?

Reply:
My understanding is that only strings can be passed to templates. How would I pass arrays?

Reply:
Let me check this

Reply:
Thanks, @VitaliyBoyko 🙏🏽